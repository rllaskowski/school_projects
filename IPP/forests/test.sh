#!/bin/bash

if [ "$#" -ne 2 ]
then
    echo "Niepoprawna ilość parametrów!"
    exit 1
fi

if ! [ -x "$1" ]
then
    echo "podaj poprawną ścieżkę do pliku wykonywalnego"
    exit 1
fi

if ! [ -d "$2" ]
then
    echo "podaj poprawną ścieżkę folderu z testami"
    exit 1
fi


for f in $2/*.in
do
    echo $f:
    valgrind --error-exitcode=15 --leak-check=full --show-leak-kinds=all --track-origins=yes --errors-for-leak-kinds=all --log-file="result.val" ./$1 < $f > result.out 2> result.err
    
    if ! grep -q "LEAK" result.val && diff ${f%in}out result.out > /dev/null && diff ${f%in}err result.err > /dev/null
    then
        echo "OK!"
        
    else
        echo "BŁĄD!"
    fi


done

rm -f result.out result.err result.val