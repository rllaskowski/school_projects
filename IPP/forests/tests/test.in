ADD drzewo1
ADD drzewo7
ADD drzewo2
ADD drzewo
ADD drzewo5
ADD drzewo2
ADD drzewo2
ADD drzewo3
ADD drzewo5
PRINT
