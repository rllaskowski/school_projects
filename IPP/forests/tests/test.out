OK
OK
OK
OK
OK
OK
OK
OK
OK
drzewo
drzewo1
drzewo2
drzewo3
drzewo5
drzewo7
